### PR TITLE
SIDM-3076 add IDAM deployments for sandbox

### PR DESCRIPTION
**Pull request by henrydobson**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SIDM-3076

### Change description ###

Add IDAM deployments for sandbox

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: k8s/sandbox/common/idam/idam-web-public.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: idam-web-public
+    version: 0.1.0
+  values:
+    java:
+      image: 'hmctspublic.azurecr.io/idam/api:preview'
+      replicas: 2
+    global:
+      environment: sandbox
+      subscriptionId:  "bf308a5c-0624-4334-8ff8-8dca9fd43783"
```
Comment (timja) [inline k8s/sandbox/common/idam/idam-web-public.yaml]: not needed anymore

File: k8s/sandbox/common/idam/idam-web-public.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+apiVersion: flux.weave.works/v1beta1
+kind: HelmRelease
+metadata:
+  name: idam-web-public
+  namespace: idam
+  annotations:
+    flux.weave.works/automated: "true"
+    flux.weave.works/tag.java: glob:prod-*
```
Comment (timja) [inline k8s/sandbox/common/idam/idam-web-public.yaml]: do you want prod or preview builds here?

File: k8s/sandbox/common/idam/idam-web-admin.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  chart:
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: idam-web-admin
+    version: 0.1.0
+  values:
+    java:
+      image: 'hmctspublic.azurecr.io/idam/web-admin:preview'
+    global:
+      environment: sandbox
+      subscriptionId:  "bf308a5c-0624-4334-8ff8-8dca9fd43783"
```
Comment (timja) [inline k8s/sandbox/common/idam/idam-web-admin.yaml]: not needed anymore

File: k8s/sandbox/common/idam/idam-web-admin.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: flux.weave.works/v1beta1
+kind: HelmRelease
+metadata:
+  name: idam-web-admin
+  namespace: idam
+  annotations:
+    flux.weave.works/automated: "true"
+    flux.weave.works/tag.java: glob:prod-*
```
Comment (timja) [inline k8s/sandbox/common/idam/idam-web-admin.yaml]: prod or preview?

File: k8s/sandbox/common/idam/idam-api.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        IDAM_SPI_FORGEROCK_IDM_ROOT: https://forgerock-idm.service.core-compute-idam-saat.internal:18443/openidm
+        IDAMHEALTHCHECK_AM_TARGET: http://forgerock-am.service.core-compute-idam-saat.internal:9292
+        IDAMHEALTHCHECK_IDM_TARGET: http://forgerock-idm.service.core-compute-idam-saat.internal:9292
+        SSL_TRUST_DNS_0: C=UK,ST=London,L=London,O=Amido,CN=*.service.core-compute-idam-saat.internal,emailAddress=devops@amido.com
+        SSL_TRUST_DNS_1: C=None,ST=None,L=None,OU=None,O=OpenDJ Self-Signed Certificate,CN=openidm-localhost
+        SSL_TRUST_HOSTNAMES_0: forgerock-idm.service.core-compute-idam-saat.internal
+        SSL_TRUST_HOSTNAMES_1: forgerock-am.service.core-compute-idam-saat.internal
+    global:
+      environment: sandbox
+      subscriptionId:  "bf308a5c-0624-4334-8ff8-8dca9fd43783"
```
Comment (timja) [inline k8s/sandbox/common/idam/idam-api.yaml]: not needed

File: k8s/sandbox/common/idam/idam-api.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        TESTING_SUPPORT_ENABLED: true
+        IDAM_SPI_FORGEROCK_AM_ROOT: https://forgerock-am.service.core-compute-idam-saat.internal:8443/openam
+        IDAM_SPI_FORGEROCK_AM_TOPLEVELHOST: forgerock-am.service.core-compute-idam-saat.internal:8443
+        IDAM_SPI_FORGEROCK_AM_JWKSURIFOROAUTH2CLIENTS: http://forgerock-am.service.core-compute-idam-saat.internal:8080/openam/oauth2/hmcts/connect/jwk_uri
+        IDAM_SPI_FORGEROCK_IDM_ROOT: https://forgerock-idm.service.core-compute-idam-saat.internal:18443/openidm
+        IDAMHEALTHCHECK_AM_TARGET: http://forgerock-am.service.core-compute-idam-saat.internal:9292
+        IDAMHEALTHCHECK_IDM_TARGET: http://forgerock-idm.service.core-compute-idam-saat.internal:9292
+        SSL_TRUST_DNS_0: C=UK,ST=London,L=London,O=Amido,CN=*.service.core-compute-idam-saat.internal,emailAddress=devops@amido.com
+        SSL_TRUST_DNS_1: C=None,ST=None,L=None,OU=None,O=OpenDJ Self-Signed Certificate,CN=openidm-localhost
+        SSL_TRUST_HOSTNAMES_0: forgerock-idm.service.core-compute-idam-saat.internal
```
Comment (timja) [inline k8s/sandbox/common/idam/idam-api.yaml]: saat or sandbox?

File: k8s/sandbox/common/idam/idam-web-public.yaml (hunk 1 above)
Comment (henrydobson) [reply k8s/sandbox/common/idam/idam-web-public.yaml]: @timja is it all of these or a subset?

Comment (timja) [reply k8s/sandbox/common/idam/idam-web-public.yaml]: subscriptionId

File: k8s/sandbox/common/idam/idam-api.yaml (hunk 6 above)
Comment (henrydobson) [reply k8s/sandbox/common/idam/idam-api.yaml]: Saat for now. Sandbox = FR 6.5. Saat = FR 5.5.

**Pull request merged**